### DISCUSSION TITLE
Does the Semantic Kernel for Java has natively support for session?

### DISCUSSION THREAD

**Discussion by marcosoliveira-jb** [microsoft/semantic-kernel#9958] [Q&A] [answered]
Hi.

I am looking to understand whether the Semantic Kernel for Java has native support for sessions to natively maintain the chat session and history automatically without my application needing to manage that for the implementation side.

Appreciate any help.
Thanks.

**Comment by sophialagerkranspandey** [comment]
Tagging @d3r3kk and @johnoliver

**Reply by d3r3kk** [reply]
Good question! I'll turn that over to @dsgrieve and @johnoliver to answer.

**Reply by marcosoliveira-jb** [reply]
@sophialagerkranspandey @dsgrieve @johnoliver do you have any updates on this one?

**Comment by dsgrieve** [comment, marked as answer]
There is no native support for sessions.

**Comment by MaxiPigna** [comment]
I believe Assistant APIs could be a viable solution; however, according to the [documentation](https://learn.microsoft.com/en-us/semantic-kernel/frameworks/agent/assistant-agent?pivots=programming-language-java), they do not appear to be supported in Java on Semantic Kernel. Have you checked the backlog to see if they will be ready soon?
